lukefi/metsi/domain/natural_processes/motti_dll_wrapper.py

```diff
@@ -0,0 +1,428 @@
+        with _maybe_chdir(self.data_dir):
+            lib.Motti4SiteInit(yy,
+                               ffi.new("float *", float(y_km)),
+                               ffi.new("float *", float(x_km)),
+                               ffi.new("float *", float(Z)),
+                               rv)
+        if rv[0] != 0:
+            raise RuntimeError(f"Motti4SiteInit failed (rv={rv[0]})")
+
+        # 2) Fill the rest (do NOT set yy.dd ourselves)
+        yy.Y = float(y_km)
+        yy.X = float(x_km)
+        yy.Z = float(Z)
+        yy.lake = float(lake)
+        yy.sea = float(sea)
+        yy.mal = float(mal)
+        yy.mty = float(self.convert_site_index(mty) if convert_mela_site else mty)
+        yy.verl = float(verl)
+        yy.verlt = float(verlt)
+        yy.alr = float(alr)
+        if year is not None:
+            yy.year = float(year)
+        yy.step = float(step)
+        # sensible defaults seen in the C wrapper
+        yy.nstorey = 1.0
+        yy.gstorey = 1.0
+
+        # allow caller to override specific scalars (including spedom/spedom2)
+        self._apply_overrides(yy, overrides)
+
+        # 3) Validate
+        nerr = ffi.new("int *")
+        err = ffi.new("int *")
+        with _maybe_chdir(self.data_dir):
+            lib.Motti4CheckYY(yy, nerr, err)
+        if nerr[0] != 0:
+            raise RuntimeError(f"Motti4CheckYY signaled problem (nerr={nerr[0]}, err={err[0]})")
+
+        return yy
+
+    def new_trees(self, trees_py: Iterable[dict]) -> Tuple[object, int]:
+        """
+        fields used: id, f, d13, h, spe, age, age13, cr, snt
+        species gets converted via Convert_Tree_Spec or 7->8, 8->9 fallback
+        """
+        ffi = self.ffi
+        yp = ffi.new("Motti4Trees *")
+        numtrees = 0
+        for i, t in enumerate(trees_py):
+            yp[0][i].id = int(t.get("id", i + 1))
+            yp[0][i].f = float(t.get("f", 0.0))
+            yp[0][i].d13 = float(t.get("d13", 0.0))
+            yp[0][i].h = float(t.get("h", 0.0))
+            yp[0][i].spe = float(self.convert_species_code(t.get("spe", 1)))
+            yp[0][i].age = float(t.get("age", 0.0))
+            yp[0][i].age13 = float(t.get("age13", 0.0))
+            yp[0][i].cr = float(t.get("cr", 0.4))
+            yp[0][i].snt = float(t.get("snt", 1))
+            yp[0][i].crerror = 0.0  # clear before growth
+            numtrees += 1
+        return yp, numtrees
+
+    # ---------- full grow (Init -> UpdateAfterImport -> loop Growth) ----------
+
+    def grow(
+            self, yy, yp, numtrees: int, step: int = 5,
+            ctrl: Optional[dict] = None, skip_init: bool = True
+        ) -> GrowthDeltas:
+        ffi, lib = self.ffi, self.lib
+        yo = ffi.new("Motti4Strata *")
+        ut = ffi.new("Motti4Saplings *")
+        kor = ffi.new("Motti4KorArray *")
+        vcr = ffi.new("Motti4VcrArray *")
+        apv = ffi.new("Motti4KorArray *")
+        fer = ffi.new("Motti4FerArray *")
```

Comment:
The naming of the variables should be more informative.

This goes for the code in overall but especially with the Motti related stuff as it has a lot of thes sort `yo`, `ut`, etc. related namings.